daffodil-core/src/main/scala/org/apache/daffodil/dpath/Expression.scala

```diff
@@ -624,7 +673,7 @@ case class WholeExpression(
 
     if (!allowCoercion) {
       val (relevantExpr: Expression, detailMsg: String) =
-        Conversion.polymorphicExpressionDiagnostics(inherentType, targetType, subExpr)
+        polymorphicExpressionErrorDetails()
```

Comment:
Is this one also essentially a no-op? Seems the only one that's needed is the DownStepExpression. If so, it probably makes sense to only move the polymorphic expression details function in DownStepExpression rather than Expression, since no other Expression classes should need this logic. It really is specific to DownStep.

Reply:
Done